subprojects/groovy-sql/src/main/java/groovy/sql/Sql.java

```diff
@@ -578,17 +578,26 @@ public static Sql newInstance(Map<String, Object> args) throws SQLException, Cla
 
         Object url = sqlArgs.remove("url");
         Connection connection;
+        LOG.fine("url = " + url);
         if (props != null) {
-            System.err.println("url = " + url);
-            System.err.println("props = " + props);
-            connection = DriverManager.getConnection(url.toString(), new Properties(props));
+            Properties propsCopy = new Properties(props);
+            connection = DriverManager.getConnection(url.toString(), propsCopy);
+            if (propsCopy.containsKey("password")) {
+                // don't log the password
+                propsCopy = new Properties(propsCopy);
```

Comment:
That makes sense, I missed that the copy was passed to the DriverManager.  Sorry for the noise.